### PR TITLE
[FIX] mrp_subcontracting: show correct view when tracked subcontracted

### PR DESCRIPTION
**Pull request by ticodoo**
To reproduce:
- create tracked product to subcontract
- create subcontract BoM with no tracked components + strict consumption
- create receipt of subcontracted product (remember to select From =
  subcontractor)
- click on `action_show_details` burger button in Operations

Expected result:
- Detailed Operations window with no move lines in it (i.e.
  stock.view_stock_move_nosuggest_operations view)
Actual Result:
- Detailed Operations window with move lines w/ 0 Done
  (i.e. stock.view_stock_move_operations view _ lines also do
  not get filled in when using the `action_assign_serial_show_details`,
  => twice as many lines as needed end up in view)

Issue appears to be that conditional was incorrectly recreated during a
minor refactoring. stock.view_stock_move_operations view
should only appear when tracked components have all been recorded
already so user can adjust their lots/sn if needed.

Task: 2695173

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp_subcontracting/models/stock_move.py [hunk 1]
```diff
@@ -179,6 +179,7 @@ def _subcontrating_can_be_record(self):
     def _get_subcontract_production(self):
         return self.filtered(lambda m: m.is_subcontract).move_orig_ids.production_id
 
+    # TODO: To be deleted, use self._get_subcontract_production()._has_tracked_component() instead
```
Comment (amoyaux) [inline addons/mrp_subcontracting/models/stock_move.py:182]: You can delete it directly no?

Comment (ticodoo) [reply addons/mrp_subcontracting/models/stock_move.py:182]: It's safe to delete a method in stable?

Comment (amoyaux) [reply addons/mrp_subcontracting/models/stock_move.py:182]: Yep better to do it in forward port

**Comment by amoyaux** [comment]
robodoo r+ rebase-ff

**Comment by adwid** [comment]
@ticodoo @amoyaux Unfortunately I think we should revert at least 6317c34808d381e54bae2431fec2de4b8503b447
With this commit, xPaths in `addons/mrp_subcontracting/views/mrp_production_views.xml` are based on the new version of `addons/mrp/views/mrp_production_views.xml` which means that all DB that already existed before the commit deployment and have not yet installed `mrp_subcontractring` will have a problem when the user installs it (because these DBs have the old version of `addons/mrp/views/mrp_production_views.xml`, so `//label[@name='bom_label']` targets nothing)

**Comment by amoyaux** [comment]
@adwid Yep you are right. Let's remove it in stable

**Pull request closed without merging**